lifetimes/estimation.py

```diff
@@ -36,6 +37,185 @@ def _print_params(self):
+
+    [1] Fader, Peter S., Bruce G.S. Hardie, and Jen Shang (2010),
+        "Customer-Base Analysis in a Discrete-Time Noncontractual Setting,"
+        Marketing Science, 29 (6), 1086-1108.
+
+    """
+    
+    def __init__(self, penalizer_coef=0.):
+        self.penalizer_coef = penalizer_coef
+
+    @staticmethod
+    def _loglikelihood(params, x, tx, T):
+
+        N = len(x)
+
+        alpha, beta, gamma, delta = params
+
+        beta_ab = special.betaln(alpha, beta)
+        beta_gd = special.betaln(gamma, delta)
+
+        indiv_loglike = (special.betaln(alpha + x, beta + T - x) - beta_ab +
+                        special.betaln(gamma, delta + T) - beta_gd)
+
+        recency_T = T - tx - 1
+
+        def _sum(x, tx, recency_T):
+
+            j = np.arange(recency_T+1)
+            return log(
+                np.sum(exp(special.betaln(alpha + x, beta + tx - x + j) - beta_ab +
+                              special.betaln(gamma + 1, delta + tx + j) - beta_gd)))
+
+        for i in np.arange(N):
+            indiv_loglike[i] = logaddexp(indiv_loglike[i],_sum(x[i], tx[i], recency_T[i]))
+
+        return indiv_loglike
+
+    @staticmethod
+    def _negative_log_likelihood(params, frequency, recency, n, n_custs, penalizer_coef=0):
+
+        return -np.sum(BetaGeoBetaBinomFitter._loglikelihood(params, frequency, recency, n) * n_custs) + penalizer_coef
+
+    def fit(self, frequency, recency, n, n_custs, verbose=False):
+        """
+        Fit the BG/BB model.
+
+        Parameters:
+            frequency: Total periods with observed transaction
+            recency: Period of most recent transaction
+            n: Number of transaction opportunities
+            n_custs: Number of customers with frequency/recency/n pattern
+
+        Returns: self
+
+        """
+
+        frequency = asarray(frequency)
+        recency = asarray(recency)
+        n = asarray(n)
+        n_custs= asarray(n_custs)
+        _check_inputs(frequency, recency, n)
+
+        params, self._negative_log_likelihood_ = _fit(self._negative_log_likelihood,
+                                                      [frequency, recency, n, n_custs, self.penalizer_coef],
+                                                      0,
+                                                      np.ones(4),
+                                                      4,
+                                                      verbose)
+        self.params_ = OrderedDict(zip(['alpha','beta','gamma','delta'], params))
+        self.data = DataFrame(vconcat[frequency, recency, n, n_custs],
+                              columns=['frequency','recency','n','n_custs'])
+
+        return self
+
+    def conditional_expected_number_of_purchases_up_to_time(self, n_star):
+        """
+        Conditional expected purchases in future time period.
+
+        The  expected  number  of  future  transactions across the next n* transaction
```

Comment:
That's the notation Fader and Hardie used for the conditional expectation. In the Pareto/NBD and BG/NBD papers they used _t_ instead. In my most recent commit I updated it to _t_ for consistency.